assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js

```diff
@@ -337,7 +356,10 @@ export default function SetupBanner( { onSubmitSuccess } ) {
 			className="googlesitekit-ga4-setup-banner"
 			title={ title }
 			ctaComponent={
-				<SpinnerButton onClick={ handleSubmitChanges }>
+				<SpinnerButton
+					onClick={ handleSubmitChanges }
+					forceProcessing={ forceProcessing }
```

Comment:
I can understand why you've added this prop. However, now that we are using a prop to specify the spinner state, it's evident that the `SpinnerButton` is more complicated than it needs to be. Bearing in mind we have only introduced it and are using it in this one case so far, I think we should remove the use of `setState` from `SpinnerButton` and always have the spinner state determined via the prop.

One additional point, I would suggest for consistency with the existing [`Spinner`](https://github.com/google/site-kit-wp/blob/f6d2929825f311086508abd110426735c65e00a6/assets/js/components/Spinner.js#L28) component that we name the prop `isSaving`.

For the sake of illustration, I've created a PR targeting this branch - please take a look and see what you think: https://github.com/google/site-kit-wp/pull/5829

Reply:
Thanks, @techanvil. I have approved and merged your changes. I have added a story to the `SpinnerButton` component to show the spinner.